endpoints/v2/manifest.py

```diff
@@ -69,7 +69,7 @@ def fetch_manifest_by_tagname(namespace_name, repo_name, manifest_ref, registry_
         )
     except RepositoryDoesNotExist as e:
         image_pulls.labels("v2", "tag", 404).inc()
-        raise NameUnknown(str(e))
+        raise NameUnknown("repository not found")
```

Comment:
When the proxy code raises a `RepositoryDoesNotExist` this will override the message it returns, making it considerably difficult for clients to understand what happened. I suggest that you change `lookup_repository` in the `OCIModel` so that it raises `RepositoryDoesNotExist("repository not found")`, then continue using the previous `str(e)` as argument to `NameUnknown` here.